features/admin.claims.v1/components/edit/local-claim/edit-basic-details-local-claims.tsx

```diff
@@ -957,18 +1056,124 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
                         hint={ t("claims:local.forms.descriptionHint") }
                         readOnly={ isSubOrganization() || isReadOnly }
                     />
-                    <Field.Checkbox
-                        ariaLabel={ t("claims:local.forms.multiValued.label") }
-                        name="multiValued"
-                        label={ t("claims:local.forms.multiValued.label") }
-                        required={ false }
-                        defaultValue={ claim?.multiValued }
-                        data-testid={ `${testId}-form-multi-valued-input` }
-                        hint={ isSystemClaim
-                            ? t("claims:local.forms.multiValuedDisabledHint")
-                            : t("claims:local.forms.multiValuedHint") }
-                        readOnly={ isSubOrganization() || isSystemClaim || isReadOnly }
+                    <Field.Dropdown
+                        ariaLabel={ t("claims:local.forms.dataType.label") }
+                        name="dataType"
+                        label={ t("claims:local.forms.dataType.label") }
+                        data-testid={ `${testId}-form-data-type-input` }
+                        hint={ t("claims:local.forms.dataType.hint") }
+                        disabled={ isSubOrganization() || isSystemClaim || isReadOnly }
+                        options={ dataTypeOptions }
+                        value={ dataType }
+                        onChange={ (
+                            event: React.SyntheticEvent<HTMLElement, Event>,
+                            data: { value: string }
+                        ) => {
+                            setSubAttributes([]);
+                            setCanonicalValues([]);
```

Comment:
Is there any specific reason to set these on onchange event of the datattype dropdown?

Reply:
When we change the dataType, I wanted to reset the states. If not those states were sent along with the API response or when switch back to a corresponding data type those states get populated.
Is there proper way to handle it.